src/test/scala/com.snowplowanalytics.sauna/utils/CommandTest.scala

```diff
@@ -0,0 +1,123 @@
+
+  case class SampleCommandData(id: String, description: String, enabled: Boolean)
+
+  implicit val sampleCommandDataReads: Reads[SampleCommandData] = Json.reads[SampleCommandData]
+
+}
+
+class CommandTest extends FunSuite with BeforeAndAfter with EitherValues with OptionValues {
+  test("extractCommand invalid JSON") {
+    val command = Json.parse(""""{}""""")
+    val result = Command.extractCommand[SampleCommandData](command)
+    assert(result.isLeft)
+  }
+
+  test("extractCommand invalid envelope") {
+    val command = Json.parse(
+      """
+        |{
+        |  "schema": "iglu:com.snowplowanalytics.sauna.commands/command/jsonschema/1-0-0",
+        |  "data": {
+        |    "envelope": {
+        |      "schema": "iglu:com.snowplowanalytics.sauna.commands/envelope/jsonschema/1-0-0",
+        |      "data": {}
+        |    },
+        |    "command": {
+        |      "schema": "iglu:com.hipchat.sauna.commands/send_room_notification/jsonschema/1-0-0",
+        |      "data": {
+        |        "id": "2a836e18-4d23-4ffb-b7b5-318a33e939eb",
+        |        "description": "Donec nec porttitor mauris.",
+        |        "enabled": true
+        |      }
+        |    }
+        |  }
+        |}
+      """.stripMargin)
+    val result = Command.extractCommand[SampleCommandData](command)
+    assert(result.isLeft)
+  }
+
+  test("extractCommand invalid data") {
+    val command = Json.parse(
+      """
+        |{
+        |  "schema": "iglu:com.snowplowanalytics.sauna.commands/command/jsonschema/1-0-0",
+        |  "data": {
+        |    "envelope": {
+        |      "schema": "iglu:com.snowplowanalytics.sauna.commands/envelope/jsonschema/1-0-0",
+        |      "data": {
+        |        "commandId": "9dadfc92-9311-43c7-9cee-61ab590a6e81",
+        |        "whenCreated": "2017-01-02T19:14:42Z",
+        |        "execution": {
+        |          "semantics": "AT_LEAST_ONCE",
+        |          "timeToLive": 1200000
+        |        },
+        |        "tags": {}
+        |      }
+        |    },
+        |    "command": {
+        |      "schema": "iglu:com.hipchat.sauna.commands/send_room_notification/jsonschema/1-0-0",
+        |      "data": {}
+        |    }
+        |  }
+        |}
+      """.stripMargin)
+    val result = Command.extractCommand[SampleCommandData](command)
+    assert(result.isLeft)
+  }
+
+  test("processEnvelope expired") {
+    val envelope = CommandEnvelope(
+      null,
+      LocalDateTime.now().minusYears(1),
+      ExecutionParams(
+        null,
+        Some(100)
+      ),
+      null)
+    val result = Command.validateEnvelope(envelope)
+    assert(result.value.contains("Command has expired"))
+  }
+
+  test("processEnvelope valid") {
+    val envelope = CommandEnvelope(
+      null,
+      LocalDateTime.now(),
+      ExecutionParams(
+        null,
+        Some(3600000)
+      ),
+      null)
+    val result = Command.validateEnvelope(envelope)
+    assert(result.isEmpty)
+  }
+}
```

Comment:
Added an extra one to `CommandTest`, but in general there's not much to work with besides a single integration test per observer/responder because there aren't many standalone testable functions. Let me know if you have any ideas though!